src/utils.ts

```diff
@@ -107,12 +107,15 @@ export function randomNumber(low: number, high: number): number {
   return Math.random() * (high - low) + low;
 }
 
-export async function timeout(ms: number) {
-  return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      resolve(null);
+export async function promiseTimeout(ms: number, promise: any) {
```

Comment:
I'm not really sure about this but it might be nice to have the promise come first in order to mimic the `setTimeout` signature that all JS devs are so used to.